nad_ch/domain/repositories.py

```diff
@@ -0,0 +1,13 @@
+from typing import List, Protocol
+from nad_ch.domain.entities import DataProvider
+
+
+class DataProviderRepository(Protocol):
+    def add(self, provider: DataProvider) -> None:
+        ...
+
+    def get_by_name(self, name: str) -> DataProvider:
+        ...
+
+    def get_all(self) -> List[DataProvider]:
```

Comment:
An `Iterable` type here would be a bit more generic.

Reply:
Makes sense since we want the interface to be as abstract as possible.